### PR TITLE
[main][feat] Update reinvest threshold logic

### PR DESCRIPTION
**Pull request by spicysquid168**
## Description
Change reinvest threshold logic to make `reinvest()` always reinvest regardless of the `pendingXXX()`.

## Why?
To avoid indefinitely pending rewards that won't get reinvested.

### CODE REVIEW

**Review by abraham-alpaca — DISMISSED** [review]

File: contracts/6/protocol/workers/pcs/PancakeswapV2Worker02.sol [hunk 1]
```diff
@@ -227,6 +230,7 @@ contract PancakeswapV2Worker02 is OwnableUpgradeSafe, ReentrancyGuardUpgradeSafe
     // 3. Withdraw all the rewards.
     masterChef.withdraw(pid, 0);
     uint256 reward = cake.balanceOf(address(this));
+    if (reward == 0) return;
```
Comment (almost-llamas) [inline contracts/6/protocol/workers/pcs/PancakeswapV2Worker02.sol:233]: Don't forget to reset approval. Or maybe just swap the step?

File: contracts/6/protocol/workers/pcs/PancakeswapV2Worker02.sol [hunk 2]
```diff
@@ -202,51 +202,58 @@ contract PancakeswapV2Worker02 is OwnableUpgradeSafe, ReentrancyGuardUpgradeSafe
...
+    uint256 _reinvestThreshold
   ) internal {
     require(_treasuryAccount != address(0), "PancakeswapV2Worker02::_reinvest:: bad treasury account");
-    // 1. Return if pendingCake <= reinvestThreshold
-    if (masterChef.pendingCake(pid, address(this)) <= reinvestThreshold) return;
 
-    // 2. Approve tokens
+    // 1. Approve tokens
     cake.safeApprove(address(router), uint256(-1));
     address(lpToken).safeApprove(address(masterChef), uint256(-1));
```
Comment (abraham-alpaca) [inline contracts/6/protocol/workers/pcs/PancakeswapV2Worker02.sol:227]: steps 1, can do things like 
```
    // 1. reward assertion 
    masterChef.withdraw(pid, 0);
    uint256 reward = cake.balanceOf(address(this));
    if (reward <= _reinvestThreshold) return 
    // 2. approve tokens 
    cake.safeApprove(address(router), uint256(-1));
    address(lpToken).safeApprove(address(masterChef), uint256(-1));
```

File: contracts/6/protocol/workers/single-asset/CakeMaxiWorker02.sol [hunk 3]
```diff
@@ -199,52 +199,60 @@ contract CakeMaxiWorker02 is OwnableUpgradeSafe, ReentrancyGuardUpgradeSafe, IWo
...
 
-    // 3. reset all reward balance since all rewards will be reinvested
+    // 2. reset all reward balance since all rewards will be reinvested
     rewardBalance = 0;
 
-    // 4. withdraw all the rewards.
+    // 3. withdraw all the rewards.
     masterChef.leaveStaking(0);
     uint256 reward = farmingToken.myBalance();
+    if (reward <= _reinvestThreshold) {
```
Comment (abraham-alpaca) [inline contracts/6/protocol/workers/single-asset/CakeMaxiWorker02.sol:230]:
```
    // 1. reward assertion 
    masterChef.leaveStaking(0);
    uint256 reward = farmingToken.myBalance();
    if (reward <= _reinvestThreshold) return 
    // 2. approve tokens 
    farmingToken.safeApprove(address(masterChef), uint256(-1));
```

File: contracts/6/protocol/workers/waultswap/WaultSwapWorker02.sol [hunk 4]
```diff
@@ -200,49 +200,59 @@ contract WaultSwapWorker02 is OwnableUpgradeSafe, ReentrancyGuardUpgradeSafe, IW
...
-    // 2. Approve tokens
+    // 1. Approve tokens
     wex.safeApprove(address(router), uint256(-1));
     address(lpToken).safeApprove(address(wexMaster), uint256(-1));
 
-    // 3. Withdraw all the rewards.
+    // 2. Withdraw all the rewards. Return if reward <= _reinvestThershold.
     wexMaster.withdraw(pid, 0, true);
     uint256 reward = wex.balanceOf(address(this));
+    if (reward <= _reinvestThreshold) {
```
Comment (abraham-alpaca) [inline contracts/6/protocol/workers/waultswap/WaultSwapWorker02.sol:229]: this too

File: contracts/6/protocol/workers/pcs/PancakeswapV2Worker02.sol [hunk 5]
```diff
@@ -202,51 +202,58 @@ contract PancakeswapV2Worker02 is OwnableUpgradeSafe, ReentrancyGuardUpgradeSafe
...
 
-    // 3. Withdraw all the rewards.
+    // 1. Withdraw all the rewards. Return if reward <= _reinvestThreshold.
```
Comment (almost-llamas) [inline contracts/6/protocol/workers/pcs/PancakeswapV2Worker02.sol:228]: Should be step 2, and and so on.

**Review by abraham-alpaca — APPROVED** [review]

**Review by almost-llamas — APPROVED** [review]

**Pull request merged**